conda/resolve.py

```diff
@@ -686,12 +652,12 @@ def generate_feature_metric(self, C):
         return eq, total
 
     def generate_removal_count(self, C, specs):
-        return {'!'+self.ms_to_v(ms): 1 for ms in specs}
+        return {'!'+self.push_MatchSpec(C, ms.name): 1 for ms in specs}
 
     def generate_package_count(self, C, missing):
-        return {self.ms_to_v(nm): 1 for nm in missing}
+        return {self.push_MatchSpec(C, nm): 1 for nm in missing}
 
-    def generate_version_metrics(self, C, specs):
+    def generate_version_metrics(self, C, specs, include0=False):
```

Comment:
I don't understand what `include0` means here, and I don't think it's actually changed to `True` anywhere, is it?